test-it/karma/javascriptinvoke/Async.Test.js

```diff
@@ -8,277 +8,797 @@ describe('A JavaScript function invoke', function () {
+                expect(viPathParser('error.status')).toBeFalse();
+                expect(viPathParser('error.code')).toBe(0);
+                expect(viPathParser('error.source')).toBeEmptyString();
+            };
+        });
+        describe('using the completion callback', function () {
+            beforeEach(function () {
+                window.SingleFunctionUnwiredReturn = function (input, jsapi) {
+                    var completionCallback = jsapi.getCompletionCallback();
+                    completionCallback(undefined);
+                };
+            });
+            it('to square a value', async function () {
+                await test();
+            });
+        });
+        describe('using promises', function () {
+            beforeEach(function () {
+                window.SingleFunctionUnwiredReturn = async function () {
+                    return undefined;
+                };
+            });
+            it('to square a value', async function () {
+                await test();
+            });
+        });
+    });
+
+    describe('errors retrieving completion callback twice', function () {
+        var error;
+        beforeEach(function () {
+            error = undefined;
+            test = async function () {
+                var viName = 'SingleFunction';
+                var runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, jsAsyncFunctionsUrl);
+                var viPathParser = vireoRunner.createVIPathParser(vireo, viName);
+                var viPathWriter = vireoRunner.createVIPathWriter(vireo, viName);
+                viPathWriter('input', 7);
+                vireoRunner.enqueueVI(vireo, viName);
+
+                const {rawPrint, rawPrintError} = await runSlicesAsync();
+                expect(rawPrint).toBeEmptyString();
+                expect(rawPrintError).toBeEmptyString();
+                expect(viPathParser('error.status')).toBeFalse();
+                expect(viPathParser('error.code')).toBe(0);
+                expect(viPathParser('error.source')).toBeEmptyString();
+                expect(viPathParser('return')).toBe(49);
+                expect(error.message).toMatch(/retrieved more than once/);
+            };
+        });
+        afterEach(function () {
+            error = undefined;
+        });
+
+        describe('using the completion callback', function () {
+            beforeEach(function () {
+                window.SingleFunction = function (input, jsapi) {
+                    var completionCallback = jsapi.getCompletionCallback();
+                    try {
+                        jsapi.getCompletionCallback();
+                    } catch (ex) {
+                        error = ex;
+                    }
+                    // TODO mraj this allows completing after an exception is thrown, this should not be allowed.
+                    // Instead if we are not resolved we should resolve immediately with an error
```

Comment:
Good catch. Can we track this somehow? Maybe as a CAR, because we do not have a US for this work.

Reply:
Made CAR 717753 for it titled "JSLI api functions that error asynchronously cause the block diagram to hang"